provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -515,6 +509,67 @@ func (c *Client) cloneVM(
 	return vm, nil
 }
 
+func (c *Client) buildConfigSpec(
+	ctx context.Context,
+	args CreateVirtualMachineParams,
+	srcVM *object.VirtualMachine,
+) (*types.VirtualMachineConfigSpec, error) {
+	var spec types.VirtualMachineConfigSpec
+	if args.Constraints.HasCpuCores() {
+		spec.NumCPUs = int32(*args.Constraints.CpuCores)
+	}
+	if args.Constraints.HasMem() {
+		spec.MemoryMB = int64(*args.Constraints.Mem)
+	}
+	if args.Constraints.HasCpuPower() {
+		cpuPower := int64(*args.Constraints.CpuPower)
+		spec.CpuAllocation = &types.ResourceAllocationInfo{
+			Limit:       &cpuPower,
+			Reservation: &cpuPower,
+		}
+	}
+	if spec.Flags == nil {
```

Comment:
Conditional's not required I think (this looks familiar, and I thought that I had removed it earlier - sorry about that)

Reply:
Oh right, removed!